test/test_pipeline.py

```diff
@@ -634,7 +655,7 @@ def test_validate_pipeline_config_invalid_pipeline_node_inputs():
         )
 
 
-def test_validate_pipeline_config_recursive_config():
+def test_validate_pipeline_config_recursive_config(reduce_windows_recursion_limit):
```

Comment:
Probably for now the fixture is needed only here, however it might be nice to enable the same fixture on all tests by moving it in `conftest.py` and marking it `autouse=True`. WDYT? Too much overhead?

Reply:
Thinking about it, we could also safely increase the scope to `session` for a minimal impact on test speed

Reply:
I'd only apply it where needed for now. Generally this seems to be only necessary on the CI either. E.g., my windows system doesn't need this fixture. So as it is only a special fix to make this test work on Windows CI I'd keep its scope as narrow as possible.